ElunaTemplate.h

```diff
@@ -339,17 +277,35 @@ class ElunaTemplate
         lua_pop(E->L, 1);
     }
 
-    template<typename C>
+    template<typename C = void>
     static void SetMethods(Eluna* E, ElunaRegister<C>* methodTable)
     {
         ASSERT(E);
-        ASSERT(tname);
         ASSERT(methodTable);
 
-        // get metatable
-        lua_pushstring(E->L, tname);
-        lua_rawget(E->L, LUA_REGISTRYINDEX);
-        ASSERT(lua_istable(E->L, -1));
+        // determine if the method table functions are global or non-global
+        bool isGlobal = false;
```

Comment:
This can just be `std::holds_alternative<int(*)(Eluna*)>(firstMethod.mfunc)` instead of visit

Reply:
I just pushed a new version of thunk, probably while you were reviewing

Reply:
The comment is on SetMethods, not thunk